src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -61,6 +61,10 @@ function warnIfInvalidElement(Component, element) {
   }
 }
 
+function isNewable(Component) {
```

Comment:
Maybe this should be called "shouldConstruct" or something like that? Functional components can be "newable" in the ECMAScript sense but shouldn't be called with new.